lib/src/barback/dartdevc/temp_environment.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:barback/barback.dart';
+import 'package:path/path.dart' as p;
+
+import '../../io.dart';
+
+/// An on-disk temporary environment for running executables that don't have
+/// a standard dart library api.
+class TempEnvironment {
```

Comment:
"Environment" is a pretty overloaded term, especially inside pub. I know it would be a pain to change, but what do you think of "ScratchSpace" or something along those lines?